Evolution/NNNN-share.md

```diff
@@ -0,0 +1,178 @@
+
+Then all sides are "drained" such that continuations are placed into the shared state and resumed when an element is available for that position.
+
+Practically this all means that a given iteration may be "behind" another and can eventually catch up (provided it is within the buffer limit).
+
+```swift
+let exampleSource = [0, 1, 2, 3, 4].async.share()
+
+let t1 = Task {
+  for await element in exampleSource {
+    if element == 0 {
+      try? await Task.sleep(for: .seconds(1))
+    }
+    print("Task 1", element)
+  }
+}
+
+let t2 = Task {
+  for await element in exampleSource {
+    if element == 3 {
+      try? await Task.sleep(for: .seconds(1))
+    }
+    print("Task 2", element)
+  }
+}
+
+await t1.value
+await t2.value
+
+```
+
+This example will print a possible ordering of the following:
+
+```
+Task 2 0
+Task 2 1
+Task 2 2
+Task 1 0
+Task 2 3
+Task 2 4
+Task 1 1
+Task 1 2
+Task 1 3
+Task 1 4
+```
+
+The order of the interleaving of the prints are not guaranteed; however the order of the elements per iteration is. Likewise in this buffering case it is guaranteed that all values are represented in the output.
+
+If the creation were altered to the following:
+
+```swift
+let exampleSource = [0, 1, 2, 3, 4].async.share(bufferingPolicy: .bufferingNewest(2))
+```
+
+The output would print the possible ordering of:
+
+```
+Task 2 0
+Task 2 1
+Task 2 2
+Task 1 0
+Task 2 4
+Task 1 3
+Task 1 4
+```
+
+Some values are dropped due to the buffering policy, but eventually they reach consistency. Which similarly works for the following:
+
+```
+let exampleSource = [0, 1, 2, 3, 4].async.share(bufferingPolicy: .bufferingOldest(2))
+```
+
+```
+Task 2 0
+Task 2 1
+Task 2 2
+Task 1 0
+Task 2 4
+Task 1 1
+Task 1 2
+```
+
+However in this particular case the newest values are the dropped elements.
+
+## Usage 
+
+It is expected that this operator will be unlike other
+
+## Effect on API resilience
+
+This is an additive API and no existing systems are changed, however it will introduce a few new types that will need to be maintained as ABI interfaces. Since the intent of this is to provide a mechanism to store AsyncSequences to a shared context the type must be exposed as ABI (for type sizing).
+
+## Alternatives considered
+
+[^BufferingPolicy] It has been considered that this particular policy would be nested inside the `AsyncShareSequence` type. However since this seems to be something that will be useful for other types it makes sense to expose it as a top level type. However if it is determined that a general form of a  buffering policy would require additional behaviors this might be a debatable placement to move back to an interior type similar to AsyncStream.
```

Comment:
FWIW we already have a type that is pretty similar called `AsyncBufferSequencePolicy`. IMO we should stay somewhat consistent across the package.